### PR TITLE
feat(apps): transpile frontend.tsx

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

To get apps to work, we need to transpile the frontend React components via a new `frontend.tsx` entrypoint.

## Changes

- Adds transpilation status fields to works started in https://github.com/PostHog/posthog/pull/9825
- Adds a step to the plugin server to transpile frontend code if it's present and not yet transpiled

## How did you test this code?

- Built new tests for the plugin server
- Tested locally by editing code and seeing the values update in the database, and the plugin get reloaded

## Final step:

- Loading the transpiled JS in the frontend

### CODE REVIEW

**Comment by mariusandra** [comment]
This is also ready for a look. There's a failing tests saying to merge the migrations, but it's a PR on a PR, so once one PR is merged, I'll rebase accordingly, depending on what gets ✅ first.

**Comment by yakkomajuri** [comment]
Changed base as I believe was the intention? I was not about to review 5000 lines :D

**Comment by Twixes** [comment]
Not sure which changes are from this PR and which from #9825, so I'll review once that other one's in, is that OK?

**Comment by yakkomajuri** [comment]
Actually can you get the branches lined up so the changes in each of the PRs are clear? @mariusandra

**Comment by mariusandra** [comment]
Indeed, something is borked. Will fix in an hour when back at my desk

**Comment by mariusandra** [comment]
Only 12 changed files now, compared to https://github.com/PostHog/posthog/pull/9825. Thus ready for a look.

File: plugin-server/src/utils/db/db.ts
Comment (mariusandra) [inline plugin-server/src/utils/db/db.ts:1933]: Note: Using `upsert` instead of `update` because in addition to `posthog_pluginsourcefile`, we can read `frontend.tsx` from github/local plugins as well. In those cases we still want to store the transpiled code somewhere.

File: plugin-server/src/utils/db/db.ts
Comment (Twixes) [inline plugin-server/src/utils/db/db.ts:1919]: This this is an enum in Python, this might be a bit easier for other devs to understand if the plugin server had a `PluginSourceFileStatus` enum too.

**Review by Twixes — APPROVED** [review]
Looks good to me (with one nit)

**Review by mariusandra — COMMENTED** [review]
@Twixes still ✅  for this?
@macobo any feedback, and/or reasons to HODL for the migration?

**Comment by mariusandra** [comment]
Gonna merge it in the end to unblock the final 🤞 PR https://github.com/PostHog/posthog/pull/9831

**Pull request merged**
